transmogrifier/sources/xml/ead.py

```diff
@@ -557,6 +493,96 @@ def get_publishers(cls, source_record: Tag) -> list[timdex.Publisher] | None:
             return [timdex.Publisher(name=publication_value)]
         return None
 
+    @classmethod
+    def get_related_items(cls, source_record: Tag) -> list[timdex.RelatedItem] | None:
```

Comment:
Reading top down, you'll likely encounter this comment before the following which came first.  But I must admit, this method broke my brain.

I had thought perhaps some of the if/else logic could be removed by targeting `<relatedmaterial>` in a list comprehension, and then the others the same way, but looking at the XML, I see now that _all_ values in the test `test_get_related_items_transforms_correctly_with_related_materials` are coming from two `<relatedmaterial>` elements.

I have no doubt it's a faithful copying of pre-existing logic, and probably not worth a rework here, but this was hard to parse for me.  That said, maybe that's just a "me" problem, and kudos to the test for thwarting what I thought were some easy tweaks!

Reply:
Appreciate the detailed breakdown!  All tracks and makes sense.

At the end of the day, this block has appeared to do what it needs to do, so I'm definitely good with it as-is even.  As dicussed here, and elsewhere, it's tricky given the variable nature of EAD.

If you _did_ want to break out into more private, sub-methods, I might propose going pretty hard in that direction, like:
  * `_get_related_from_relatedmaterial()`
  * `_get_related_from_altformavail()`
  * `_get_related_from_separatedmaterial()`

Perhaps this is a throwback to a comment I made on one of @ehanson8's PRs about a variable name that smushed up two words.  I'm appreciating now that it was probably because the element was named as such, so it feels accurate; and I agree.  Apologies @ehanson8 if I misssed that nuance at the time!  I feel as though, now, I would read these methods -- particularly when they were right next to one another when being called -- and understand their naming is reflective of the XML elements, e.g.

```python
related_items = []
related_items.extend(_get_related_from_relatedmaterial())
related_items.extend(_get_related_from_altformavail())
related_items.extend(_get_related_from_separatedmaterial())
```

But... to your points @jonavellecuerdo about other similar methods not breaking it out into sub-methods like this, I think that's valid.  

I've been appreciating during these PR reviews of the thoughtful consideration to the goal of the refactoring to fields methods vs general refactoring of logic that has not been revisited in awhile.  I would vote for continuing to err on the side of _not_ making too many sweeping changes during this first pass.  Which is not to say changes that have been made were a bad idea, quite the contrary they all seemed good, just that if it gets thorny without a clear, swift improvement, maybe better to just leave for now.  Just my two cents.

Reply:
Please see updated tests! Also decided to rename private method to `_get_related_from_relatedmaterial`. Although long, I concur that it is a great naming convention that clearly describes the XML element source!